yambo-front/src/components/CardCreate.tsx

```diff
@@ -249,13 +292,15 @@ const CardCreate: FC = () => {
       definition: ""
     });
     setDisableTab(!disableTab);
-    setCardPlaceholder("Add another card...");
+    setCardPlaceholder("Add a card...");
     setCardView("front");
   };
 
-  function handleSubmit(e: MouseEvent) {
+  function handleLookup(e: MouseEvent) {
     e.preventDefault();
 
+    setLoader(true);
+
     try {
       //Use for actual api call
       const fetchData = async () => {
```

Comment:
After the fetch call, you can check whether or not the response returned with a status of 200:
        if (response.status === 200) {
then continue setting the card data. And you could move the spinner outside of the         if (response.status === 200) {} so you only have to call setLoader(false) once.